CMakeLists.txt

```diff
@@ -151,7 +154,7 @@ set(IREE_EXTERNAL_TOOLING_MODULES "" CACHE STRING "")
 #-------------------------------------------------------------------------------
 
 set(IREE_COMPILER_PLUGIN_PATHS "" CACHE STRING "Paths to external compiler plugins")
-set(IREE_COMPILER_BUILTIN_PLUGIN_PATHS "compiler/plugins")
+list(APPEND IREE_COMPILER_BUILTIN_PLUGIN_PATHS "compiler/plugins")
```

Comment:
Hmm... 

Currently:
* IREE sets `IREE_COMPILER_BUILTIN_PLUGIN_PATHS`
* Users can set `IREE_COMPILER_PLUGIN_PATHS` for external plugins

This PR:
* Downstream ("super-project") sets `IREE_COMPILER_BUILTIN_PLUGIN_PATHS`
* IREE appends to `IREE_COMPILER_BUILTIN_PLUGIN_PATHS`
* Users can set `IREE_COMPILER_PLUGIN_PATHS` for external plugins

Seems like downstream projects could prepopulate `IREE_COMPILER_PLUGIN_PATHS` instead of needing to touch `IREE_COMPILER_BUILTIN_PLUGIN_PATHS`? We could also rename to `IREE_EXTERNAL_COMPILER_PLUGIN_PATHS` to match `IREE_EXTERNAL_TOOLING_MODULES` and `IREE_EXTERNAL_HAL_DRIVERS`...

Reply:
It is very bad ergonomics for a super-project to be touching the user-level cache variable (which the user may want to use otherwise). And then you get into tug of wars on who is forcing who's variable. I prefer that lexically scoped projects are able to either work with variables or global properties vs the cache variable tug of ware.

Reply:
Agreed. For this case I'm wondering if the core project would still find it useful to know which paths are coming from its own directory and which are coming from a super-project. We _could_ add another variable if so. Otherwise, the comments are probably sufficient.